### PR TITLE
Remove bsm/sarama-cluster library since it's deprecated

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fixes https://github.com/knative/eventing/issues/944

### CODE REVIEW

File: kafka/channel/pkg/dispatcher/consumer_handler.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			zap.Any("offset", message.Offset),
+		)
+
+		knativeMessage := fromKafkaMessage(message)
+		err := consumer.dispatcherFunc(knativeMessage, &consumer.sub)
+		if err != nil {
+			consumer.logger.Warn("Got error trying to dispatch knativeMessage", zap.Error(err))
+		}
+
+		session.MarkMessage(message, "") // Mark knativeMessage as processed
```
Comment (matzew) [inline kafka/channel/pkg/dispatcher/consumer_handler.go]: let's only `MarkMessage` really when the actual msg got dispatched,

see: 
https://github.com/knative/eventing-contrib/blob/master/kafka/source/pkg/adapter/adapter.go#L89-L95

**Comment by matzew** [comment]
@slinkydeveloper  can you also run `./hack/update-deps.sh` after code changes?

File: kafka/channel/pkg/utils/util.go [hunk 2]
```diff
@@ -71,18 +68,6 @@ func GetKafkaConfig(path string) (*KafkaConfig, error) {
 		return nil, fmt.Errorf("missing key %s in configuration", BrokerConfigMapKey)
 	}
 
-	config.ConsumerMode = cluster.ConsumerModeMultiplex
-	if mode, ok := configMap[ConsumerModeConfigMapKey]; ok {
-		switch strings.ToLower(mode) {
-		case ConsumerModeMultiplexConsumerValue:
```
Comment (matzew) [inline kafka/channel/pkg/utils/util.go]: @slinkydeveloper   remove those definitions:

https://github.com/knative/eventing-contrib/blob/9c471692f6cb04e2c56729b4fb94fdca135df526/kafka/channel/pkg/utils/util.go#L28-L30

File: kafka/channel/pkg/dispatcher/dispatcher.go [hunk 3]
```diff
@@ -59,41 +59,40 @@ type TopicFunc func(separator, namespace, name string) string
 type KafkaDispatcherArgs struct {
 	ClientID     string
 	Brokers      []string
-	ConsumerMode cluster.ConsumerMode
```
Comment (matzew) [inline kafka/channel/pkg/dispatcher/dispatcher.go]: @slinkydeveloper  the `consumerMode` is set via this config map:
https://github.com/knative/eventing-contrib/blob/master/kafka/channel/config/400-kafka-config.yaml#L28

please update yaml and remove code that reads it.

Also, this needs to be release noted!

Comment (slinkydeveloper) [reply kafka/channel/pkg/dispatcher/dispatcher.go]: there is any crd to update too?

**Pull request closed without merging**